src/com/qamp/HarisJasarevic/homeworks/homework2/task3/Robot.java

```diff
@@ -0,0 +1,74 @@
+package com.qamp.HarisJasarevic.homeworks.homework2.task3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Robot {
+
+    private int x;
+    private int y;
+    private int robotMoves;
+    private List<String> robotPositions;
+
+    public Robot() {
+        this.x = 0;
+        this.y = 0;
+    }
+
+    public Robot(final int x, final int y) {
+        this.x = x;
+        this.y = y;
+        this.robotMoves = 0;
+        this.robotPositions = new ArrayList<>();
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesSouth () {
+        this.y--;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesNorth () {
+        this.y++;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesEast () {
+        this.x++;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
```

Comment:
This line `this.robotPositions.add("(" + this.x + "," + this.y + ")");` repeats 4 times, should be extracted into a method and reused.

You might already have such method :), `getRobotPosition`